packages/react-reconciler/src/ReactFiberWorkLoop.new.js

```diff
@@ -412,28 +415,76 @@ export function requestUpdateLane(
           : NoLanes;
     }
 
-    lane = findTransitionLane(
+    return findTransitionLane(
       transitionLanePriority,
       currentEventWipLanes,
       currentEventPendingLanes,
     );
-  } else {
-    // TODO: If we're not inside `runWithPriority`, this returns the priority
-    // of the currently running task. That's probably not what we want.
-    const schedulerPriority = getCurrentPriorityLevel();
+  }
 
-    if (
-      // TODO: Temporary. We're removing the concept of discrete updates.
-      (executionContext & DiscreteEventContext) !== NoContext &&
-      schedulerPriority === UserBlockingSchedulerPriority
-    ) {
-      lane = findUpdateLane(InputDiscreteLanePriority, currentEventWipLanes);
-    } else {
-      const lanePriority = schedulerPriorityToLanePriority(schedulerPriority);
-      lane = findUpdateLane(lanePriority, currentEventWipLanes);
+  // TODO: Remove this dependency on the Scheduler priority.
```

Comment:
I refactored this method again so that the feature flag only opts into the new behavior if the scheduler priority matches the update lane priority. If it doesn't match, we log the mismatch and fall back to the old behavior.

This will allow us to fix all of the warnings incrementally, while also turning on the feature flag to test the experiment, instead of needing to fix all the issues before testing.

Let me know what you think 👍

Reply:
I think the only thing it should do is fire the warning. The goal of this first step is that no behavior should change. We're just shuffling around an implementation detail.

Reply:
If the flag changes the semantics, then we can't turn it on safely in www, which goes against the purpose of using the warning to find the places that need to change.

Reply:
Sounds good - should the warning be behind the feature flag?

Reply:
I got the answer below, thanks 👍